### PR TITLE
Port all test cases to Jest tests

### PR DESCRIPTION
**Pull request by saikat-samanta**
Need to refactor:

- [x] One test is failing in `ffis.test.ts`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -37,8 +37,8 @@
     "build": "tsc",
     "postbuild": "cp src/libinchi.so.1.06.00 dist/tsc/src",
     "serve": "node dist/tsc/src/server.js",
-    "test:headers": "node dist/tsc/tests/headers.test.js",
-    "test:ffis": "node dist/tsc/tests/ffis.test.js",
+    "test:headers": "jest tests/headers.test.ts",
+    "test:ffis": "jest tests/ffis.test.ts",
```
Comment (maneetgoyal) [inline package.json:41]: Convert it into single command: `test: ....`

**Comment by maneetgoyal** [comment]
@saikat-samanta Tests are failing.

File: tests/ffis.test.ts [hunk 2]
```diff
@@ -1,122 +1,148 @@
...
+        
+          5  5  0  0  0  0  0  0  0  0999 V2000
+            1.8326    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+            2.2652    1.3315    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+            1.1326    2.1544    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+            0.0000    1.3315    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+            0.4326    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+          1  2  1  0  0  0  0
+          2  3  1  0  0  0  0
+          3  4  1  0  0  0  0
```
Comment (maneetgoyal) [inline tests/ffis.test.ts:68]: Why were the mol files changed?

**Review by maneetgoyal — CHANGES_REQUESTED** [review]

File: tests/ffis.test.ts (hunk 2 above)
Comment (saikat-samanta) [reply tests/ffis.test.ts:68]: @maneetgoyal sir, previous one was failing. I think data was wrong

**Comment by saikat-samanta** [comment]
> @saikat-samanta Tests are failing.

@maneetgoyal sir, test are passing in local system
Screenshot_3

File: tests/ffis.test.ts [hunk 3]
```diff
@@ -1,122 +1,148 @@
...
-    0.0000   -0.0000   -0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
-M  END
-`,
-    "",
-    output2.ref()
-  ),
-  0
-);
-strict.equal(output2.szInChI, "InChI=1S/CH4/h1H4");
+    const output2 = inchi_Output({
```
Comment (maneetgoyal) [inline tests/ffis.test.ts:48]:
```suggestion
    const output2 = new inchi_Output({
```

File: tests/ffis.test.ts [hunk 4]
```diff
@@ -1,122 +1,148 @@
...
+      szInChI: "1S/C5H10/c1-2-4-5-3-1/h1-5H2",
```
Comment (maneetgoyal) [inline tests/ffis.test.ts:49]:
```suggestion
      szInChI: "",
```

File: tests/ffis.test.ts [hunk 5]
```diff
@@ -1,122 +1,148 @@
...
-M  END
-`,
-    "",
-    output3
-  ),
-  0
-);
-strict.equal(output3.deref().szInChI, "InChI=1S/C2H6O/c1-2-3/h3H,2H2,1H3");
+    const output3 = refNAPI.alloc(inchi_Output, {
+      szInChI: "1S/C5H10/c1-2-4-5-3-1/h1-5H2",
```
Comment (maneetgoyal) [inline tests/ffis.test.ts:80]:
```suggestion
      szInChI: "",
```

File: tests/ffis.test.ts [hunk 6]
```diff
@@ -1,122 +1,148 @@
...
+          3  4  1  0  0  0  0
+          4  5  1  0  0  0  0
+          5  1  1  0  0  0  0
+        M  END
+        `,
+        "",
+        output3
+      )
+    ).toBe(2);
+    expect(output3.deref().szInChI).toBe("1S/C5H10/c1-2-4-5-3-1/h1-5H2");
```
Comment (maneetgoyal) [inline tests/ffis.test.ts:108]:
```suggestion
    expect(output3.deref().szInChI).toBe("InChI=1S/C5H10/c1-2-4-5-3-1/h1-5H2");
```

Should `InChI=` be used? In some cases it is being used. Try to figure why is this inconsistency?

**Comment by maneetgoyal** [comment]
>@maneetgoyal sir, test are passing in local system

@saikat-samanta Try to debug the CI. Local is passing on my system too.

**Comment by saikat-samanta** [comment]
> > @maneetgoyal sir, test are passing in local system
> 
> @saikat-samanta Try to debug the CI. Local is passing on my system too.

@maneetgoyal sir, okay

File: package.json [hunk 7]
```diff
@@ -37,9 +37,7 @@
     "build": "tsc",
     "postbuild": "cp src/libinchi.so.1.06.00 dist/tsc/src",
     "serve": "node dist/tsc/src/server.js",
-    "test:headers": "node dist/tsc/tests/headers.test.js",
-    "test:ffis": "node dist/tsc/tests/ffis.test.js",
-    "test": "yarn test:headers && yarn test:ffis",
+    "test": "jest tests/ffis.test.ts && jest tests/headers.test.ts",
```
Comment (maneetgoyal) [inline package.json:40]: - Try to figure out why the earlier command wasn't working.
- Look into https://jestjs.io/docs/cli#--runinband also.

**Comment by saikat-samanta** [comment]
Ref: https://codewithhugo.com/run-jest-sequentially/

File: tests/ffis.test.ts [hunk 8]
```diff
@@ -1,122 +1,148 @@
...
+          3  4  1  0  0  0  0
+          4  5  1  0  0  0  0
+          5  1  1  0  0  0  0
+        M  END
+        `,
+        "",
+        output2.ref()
+      )
+    ).toBe(2);
+    expect(output2.szInChI).toBe("");
```
Comment (maneetgoyal) [inline tests/ffis.test.ts:77]: This is unexpected! It should be a non empty string. Revert all the mol files.

**Review by maneetgoyal — CHANGES_REQUESTED** [review]

File: tests/ffis.test.ts (hunk 8 above)
Comment (saikat-samanta) [reply tests/ffis.test.ts]: done 👍

Comment (maneetgoyal) [reply tests/ffis.test.ts]: Not done yet. Don't abruptly change the code. Try to understand what it means first. 

Firstly, you need to revert all the mol file strings. This is not done yet.

Secondly, the input `szInChI` should be empty but the output `szInChI` should be non-empty.

File: tests/ffis.test.ts [hunk 9]
```diff
@@ -1,122 +1,148 @@
...
+            2.2652    1.3315    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+            1.1326    2.1544    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+            0.0000    1.3315    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+            0.4326    0.0000    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+          1  2  1  0  0  0  0
+          2  3  1  0  0  0  0
+          3  4  1  0  0  0  0
+          4  5  1  0  0  0  0
+          5  1  1  0  0  0  0
+        M  END
```
Comment (maneetgoyal) [inline tests/ffis.test.ts:102]: This is an example of mol file strings that needs to be changed!

Comment (saikat-samanta) [reply tests/ffis.test.ts]: @maneetgoyal sir, done. But tests are failing now.

Comment (maneetgoyal) [reply tests/ffis.test.ts]: That's okay, let me check

**Review by maneetgoyal — APPROVED** [review]

File: tests/ffis.test.ts (hunk 9 above)
Comment (maneetgoyal) [reply tests/ffis.test.ts]: @saikat-samanta The old files are working now. Please merge after CI is green.

**Pull request merged**